From c3f7d36b992a3cbcee3386dea85720f3cb04e1ff Mon Sep 17 00:00:00 2001
From: Song Bing <b06498@freescale.com>
Date: Fri, 27 Mar 2015 13:39:43 +0800
Subject: [PATCH] videoencoder: Keep sticky events around when doing a soft
 reset

The current code will first discard all frames, and then tries to copy
all sticky events from the (now discarded) frames. Let's change the order.

https://bugzilla.gnome.org/show_bug.cgi?id=746865

Upstream-Status: Accepted 

---
 gst-libs/gst/video/gstvideoencoder.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gst-libs/gst/video/gstvideoencoder.c b/gst-libs/gst/video/gstvideoencoder.c
index 614ba2d..4c5b111 100644
--- a/gst-libs/gst/video/gstvideoencoder.c
+++ b/gst-libs/gst/video/gstvideoencoder.c
@@ -340,10 +340,6 @@ gst_video_encoder_reset (GstVideoEncoder * encoder, gboolean hard)
 
   priv->drained = TRUE;
 
-  g_list_foreach (priv->frames, (GFunc) gst_video_codec_frame_unref, NULL);
-  g_list_free (priv->frames);
-  priv->frames = NULL;
-
   priv->bytes = 0;
   priv->time = 0;
 
@@ -392,6 +388,10 @@ gst_video_encoder_reset (GstVideoEncoder * encoder, gboolean hard)
         encoder->priv->current_frame_events);
   }
 
+  g_list_foreach (priv->frames, (GFunc) gst_video_codec_frame_unref, NULL);
+  g_list_free (priv->frames);
+  priv->frames = NULL;
+
   GST_VIDEO_ENCODER_STREAM_UNLOCK (encoder);
 
   return ret;
-- 
1.7.9.5

